src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -354,7 +354,10 @@ def recursive_python_sweep(
                     setattr(
                         sequence[kdx], sweeper_parameter.name.lower(), values[jdx][idx]
                     )
-                elif sweeper is rfsoc.Parameter.DELAY:
+                    if sweeper_parameter is rfsoc.Parameter.DURATION:
+                        for pulse_idx in range(kdx + 1, len(sequence)):
+                            sequence[pulse_idx].start = sequence[pulse_idx - 1].duration
```

Comment:
This is slightly better than the former patch, because it is less hard-coded for the Rabi.
However, it is still a quick and dirty patch, because it would advance even pulses that should be played in parallel.
E.g.:
```
drive (start=0, duration swept)
flux (start=0)
readout (start=drive.duration)
```
not only the readout pulse is advanced, but even the flux.

The proper fix for this is one of these:
1. diagonal (simultaneous) sweeper, and sweep not only the duration of the drive, but even the measurement start
2. add a delay on the readout line, and you can sweep the duration of the drive and readout delay in the same sweeper (this should be fixed in 0.2, if we'll have simultaneous sweepers, better to use one pulse per sweeper @stavros11)
3. add an `align()` between the drive and readout pulse

Since the diagonal sweepers and aligns are not yet supported by Qibolab, the only viable solution is 2., assuming you could mock a readout delay (like adding a readout pulse with 0 amplitude, and then discarding the result), but it would require to change the Qibocal routine.

To be fair, the Qibocal routine should be changed in any case, since the sweeper is just on the duration, and it is never specified that you should advance the reaodut pulse as well, so it might be up to the drivers to push measurements at the end, and it would be wrong (because in general, you should be free of playing the readout wherever you wish).

Reply:
> To be fair, the Qibocal routine should be changed in any case, since the sweeper is just on the duration, and it is never specified that you should advance the reaodut pulse as well, so it might be up to the drivers to push measurements at the end, and it would be wrong (because in general, you should be free of playing the readout wherever you wish).

@andrea-pasquale is it correct? In case, I'll open an issue on Qibocal

Reply:
> Right now qibocal is leveraging the `start` attribute to hint at the fact that the readout should be pushed.

The problem is that is not truly a hint, since the official specification of that would be that is exactly starting at that time (the `.finish` of the previous pulse) that is a number even in the case in which the other pulse is swept.
Pulses swept are normal pulses in the sequence, then the sweeper will later change its attributes, but there is no link with other pulses with *accidentally* the same values (or related ones) for same or other properties (like `pulse1.finish == pulse2.start` is a relation for the original sequence, but it is not checked in any way when the sweeper will modify the value of `pulse1.duration`, and thus its end).